addons/binding/org.openhab.binding.sinope/src/main/java/org/openhab/binding/sinope/handler/SinopeThermostatHandler.java

```diff
@@ -0,0 +1,279 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.sinope.handler;
+
+import java.io.IOException;
+import java.net.UnknownHostException;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.sinope.SinopeBindingConstants;
+import org.openhab.binding.sinope.internal.config.SinopeConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.tulip.sinope.core.SinopeDataReadRequest;
+import ca.tulip.sinope.core.SinopeDataWriteRequest;
+import ca.tulip.sinope.core.appdata.SinopeHeatLevelData;
+import ca.tulip.sinope.core.appdata.SinopeOutTempData;
+import ca.tulip.sinope.core.appdata.SinopeRoomTempData;
+import ca.tulip.sinope.core.appdata.SinopeSetPointModeData;
+import ca.tulip.sinope.core.appdata.SinopeSetPointTempData;
+import ca.tulip.sinope.core.internal.SinopeDataAnswer;
+import ca.tulip.sinope.util.ByteUtil;
+
+/**
+ * The {@link SinopeThermostatHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Pascal Larin - Initial contribution
+ */
+public class SinopeThermostatHandler extends BaseThingHandler implements ThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(SinopeThermostatHandler.class);
+
+    private SinopeGatewayHandler gatewayHandler;
+
+    public SinopeThermostatHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        Channel channel = getThing().getChannel(channelUID.getId());
+        SinopeGatewayHandler gateway = getSinopeGatewayHandler();
+        gateway.stopPoll();
+        try {
+            if (channel != null && SinopeBindingConstants.CHANNEL_SETTEMP.equals(channelUID.getId())
+                    && command instanceof DecimalType) {
+                setSetpointTemp(this, ((DecimalType) command).floatValue());
+            }
+
+            if (channel != null && SinopeBindingConstants.CHANNEL_SETMODE.equals(channelUID.getId())
+                    && command instanceof StringType) {
```

Comment:
If you want an integer, i.e. a number, why do you expect a `StringType` here instead of a `DecimalType`?